mat64/format.go

```diff
@@ -9,15 +9,66 @@ import (
 	"strconv"
 )
 
-// Format prints a pretty representation of m to the fs io.Writer. The format character c
+// Formatted returns a fmt.Formatter for the matrix m using the given options.
+func Formatted(m Matrix, options ...FormatOption) fmt.Formatter {
+	f := formatter{
+		matrix: m,
+		dot:    '.',
+	}
+	for _, o := range options {
+		o(&f)
+	}
+	return f
+}
+
+type formatter struct {
+	matrix Matrix
+	prefix string
+	margin int
+	dot    byte
+}
+
+// FormatOption is a functional option for matrix formatting.
+type FormatOption func(*formatter)
+
+// Prefix sets the formatted prefix to the string p. Prefix is a string that is prepended to
+// each line of output.
+func Prefix(p string) FormatOption {
+	return func(f *formatter) { f.prefix = p }
+}
+
+// Margin sets the format margin to m. If m is greater than zero, the margin option indicates
```

Comment:
"Margin" doesn't have quite the right meaning for me.  Practically it's more like "corners" than margins.  Conceptually, it's more of an "exerpt."  "Take" would have meaning for programmers.  (TakeCorners?)  But it's a small thing.  Especially if the word margin is used like this in other software, then margin is fine.